source/nodejs/adaptivecards-designer/src/designer-peers.ts

```diff
@@ -3313,4 +3334,123 @@ export class TablePeer extends TypedCardElementPeer<Adaptive.Table> {
+
+        commands.push(
+            new PeerCommand(
+                {
+                    name: "Add a page",
+                    isPromotable: false,
+                    execute: (command: PeerCommand, clickedElement: HTMLElement) => {
+                        let page = new Adaptive.CarouselPage();
+
+                        (this.cardElement as Adaptive.Carousel).addPage(page);
+
+                        this.updateChildren();
+                    }
+                }
+            )
+        )
+    }
+
+    initializeCardElement() {
+        super.initializeCardElement();
+
+        (this.cardElement as Adaptive.Carousel).addPage(new Adaptive.CarouselPage());
+    }
+
+    populatePropertySheet(propertySheet: PropertySheet, defaultCategory: string = PropertySheetCategory.DefaultCategory) {
+        super.populatePropertySheet(propertySheet, defaultCategory);
+
+        propertySheet.remove(CardElementPeer.isVisibleProperty,
+            ContainerPeer.bleedProperty, 
+            ContainerPeer.styleProperty);
+
+        propertySheet.add(
+            defaultCategory,
+            CarouselPeer.timerProperty);
+    }
+
+    canDrop(peer: DesignerPeer) {
+        return false;
+    }
+
+    isDraggable(): boolean {
+        return false;
+    }
+
+    attachOnPageChange() {
+        (this.cardElement as Adaptive.Carousel).onPageChanged = (activeIndex: number, loopIndex: number) => {
+            const carouselElement = this.cardElement as Adaptive.Carousel;
+
+            // TODO: Refactor to use realIndex
+            if (activeIndex === 0) {
+                // Index 0 is a duplicate slide, and we should slide to the end
+                carouselElement.carousel?.slideTo(this.getChildCount())
+            } else if (activeIndex === (this.getChildCount() + 1)) {
+                // Last index is a duplicate slide, and we should slide to the beginning
+                carouselElement.carousel?.slideTo(1);
+            } else if (carouselElement.currentIndex != (activeIndex - loopIndex)) {
+                // Valid index, rerender the card
+                carouselElement.currentIndex = activeIndex - loopIndex;
+                this.designerSurface.render();
+            }
+
+            // Since the carousel changed, we will need to update the designer surface layout
+            this.designerSurface.needsLayoutUpdate = true;
+        };
+    }
+}
+
+export class CarouselPagePeer extends ContainerPeer {
+
+    isDraggable(): boolean {
+        return false;
+    }
+
+    getBoundingRect(): Rect {
+        const boundingRect = super.getBoundingRect();
+
+        const carousel = this.cardElement.parent as Adaptive.Carousel;
+
+        if (carousel) {
+            const containerClientRect = carousel.carouselPageContainer.getBoundingClientRect();
+
+            const leftOffset = carousel.renderedElement.offsetLeft + carousel.carouselPageContainer.offsetLeft;
+
+            boundingRect.right = leftOffset + boundingRect.width;
```

Comment:
shift the order and remove leftOffset;

boundingRect.left = carousel.renderedElement.offsetLeft + carousel.carouselPageContainer.offsetLeft;
boundingRect.right = boundingRect.left + boundingRect.width;